picard/ui/coverartbox.py

```diff
@@ -127,13 +127,9 @@ def __eq__(self, other):
         else:
             return True
 
-    @staticmethod
-    def dragEnterEvent(event):
-        event.acceptProposedAction()
-
-    @staticmethod
-    def dragMoveEvent(event):
-        event.acceptProposedAction()
+    def dragEnterEvent(self, event):
+        event.setDropAction(QtCore.Qt.DropAction.CopyAction)
```

Comment:
According to the QT documentation at https://doc.qt.io/qt-6/qt.html#DropAction-enum existing drop actions can be:

Constant | Value | Description
-- | -- | --
Qt::CopyAction | 0x1 | Copy the data to the target.
Qt::MoveAction | 0x2 | Move the data from the source to the target.
Qt::LinkAction | 0x4 | Create a link from the source to the target.
Qt::ActionMask | 0xff |  
Qt::IgnoreAction | 0x0 | Ignore the action (do nothing with the data).
Qt::TargetMoveAction | 0x8002 | On Windows, this value is used when the ownership of the D&D data should be taken over by the target application, i.e., the source application should not delete the data. On X11 this value is used to do a move. TargetMoveAction is not used on the Mac.

I still believe that we should check the existing DropAction and only change it if it is not `Qt::IgnoreAction` on the basis that changing a MoveAction/LinkAction to a CopyAction seems reasonable, but (just in case we ever get one) changing an IgnoreAction to a CopyAction seems wrong. (I am not sure how an IgnoreAction might happen, but I guess that starting a Move and then aborting it would result in an IgnoreAction in order to inform the receiving app that the Drag&Drop which they were already told to expect is no longer going to happen. If this is the basis for an IgnoreAction, we would not IMO want to change it to a CopyAction.)

Reply:
Maybe we should handle this event, but before we change anything in this regard I would like to know how this event can actually be triggered and how it is supposed to be handled. The Qt5 documentation is pretty sparse here.

Looking through Qt's code samples it seems this action sometimes gets handled by not performing any action but accepting the action, e.g. https://github.com/qt/qtbase/blob/33f7eb44ff33e0f8f516a9e8076dfdb6098cfc6f/tests/manual/examples/widgets/itemviews/puzzle/piecesmodel.cpp#L108-L109

But it's all really vague. If you have more information on how this event is used and under which circumstances it triggers we can investigate how to handle it in the different scenarios where drops are supported.

For the itemviews it likely is not a problem anyway, as anything other then CopyAction and MoveAction is not accepted.

Reply:
I added https://github.com/metabrainz/picard/pull/2271 with how I think it should be handled based on the Qt examples. But I don't see how to verify this is the correct way to handle the ignore message. I very much would like to have some kind of reproducer for this before merging that code.